lib/capybara/query.rb

```diff
@@ -43,6 +43,10 @@ def matches_filters?(node)
         regexp = options[:text].is_a?(Regexp) ? options[:text] : Regexp.escape(options[:text].to_s)
         return false if not node.text(visible).match(regexp)
       end
+      if options[:href]
```

Comment:
Turns out that the changes I made to the `link` selector made this filter unnecessary anyway.